### ISSUE TITLE
Could I run BytePS without using pipeline and scheduler?

### ISSUE THREAD

**Issue by wizard1203** [bytedance/byteps#227]
Hi, 
I want to test the performance of vanilla Parameter Server. So I want to run BytePS without using pipeline and scheduler.

I know I can directly turn off the scheduler. But for pipeline, can I directly turn it off, or I need to change codes? 

If I need to change codes, is there any difference in the pipeline part implemented in pytorch and tensorflow? Because some frameworks implement pipeline by themselves. Dose BytePS exploit their original pipeline API or implement pipeline by BytePS itself.

**Comment by bobzhuyb** [comment]
> I know I can directly turn off the scheduler. But for pipeline, can I directly turn it off, or I need to change codes?

You have to do some coding. 

> If I need to change codes, is there any difference in the pipeline part implemented in pytorch and tensorflow? Because some frameworks implement pipeline by themselves. Dose BytePS exploit their original pipeline API or implement pipeline by BytePS itself.

Which pipeline are you referring to? The computation-communication pipeline, or push-pull pipeline? The former is in the engine of pytorch and TF, the latter is in BytePS and is agnostic to frameworks.

**Comment by wizard1203** [comment]
> Which pipeline are you referring to? The computation-communication pipeline, or push-pull pipeline? The former is in the engine of pytorch and TF, the latter is in BytePS and is agnostic to frameworks.

Thank you. I want to remove all of them. Emm, this may take some time.

**Comment by bobzhuyb** [comment]
> > Which pipeline are you referring to? The computation-communication pipeline, or push-pull pipeline? The former is in the engine of pytorch and TF, the latter is in BytePS and is agnostic to frameworks.
> 
> Thank you. I want to remove all of them. Emm, this may take some time.

I suggest you not waste time on this, no matter in the sense of developing practical systems or research. I don't see any point reverting back to sub-optimal solutions.

Anyways, I am closing this issue since it's neither a bug or a feature request.
